### DISCUSSION TITLE
Replicate the behavior of vi's `a`

### DISCUSSION THREAD

**Discussion by ispringle** [meow-edit/meow#497] [Q&A] [answered]
I'm liking meow. I've spend a few years now using doom and evil and before that I'd spent my life in vim. I'm trying to get closer to emacs and meow seems great for that. One thing I seem to be missing though is the default behavior of `a` in vi. Normally if I my cursor at `w[o]rd` and I type `a` from NORMAL mode I'd expect to enter INSERT mode at `wo|rd`. In meow `a` acts the same as `i` (when not dealing with a selection) and inserts my cursor at `w|ord`. Is there a way to get `a` to replicate vi in NORMAL mode when not working with a selection?

**Comment by DogLooksGood** [comment, marked as answer]
Well, you have to write your own command for such purpose. 

```
(defun my-meow-append ()
  "Move to the end of selection, switch to INSERT state."
  (interactive)
  (if meow--temp-normal
      (progn
        (message "Quit temporary normal mode")
        (meow--switch-state 'motion))
    (if (not (region-active-p))
        (when (and (not (use-region-p))
                   (< (point) (point-max)))
          (forward-char 1))
      (meow--direction-forward)
      (meow--cancel-selection))
    (meow--switch-state 'insert)))
```
Modified from the original `meow-append`.

**Reply by ispringle** [reply]
thank you for the help!
